### PR TITLE
Paginate ga profiles

### PR DESCRIPTION
**Pull request by bfeigin**

### CODE REVIEW

File: lib/legato/management/finder.rb [hunk 1]
```diff
@@ -14,10 +15,31 @@ def all(user, path=default_path)
...
-        json = user.access_token.get(base_uri + path).body
-        items = MultiJson.decode(json).fetch('items', [])
-        items.map {|item| new(item, user)}
+        collect_items(user, uri).map do |item|
+          new(item, user)
+        end
       end
+
+      private
+        def items_key; "items"; end
```
Comment (GreyKn) [inline lib/legato/management/finder.rb]: Why not consts for these?

File: lib/legato/management/finder.rb [hunk 2]
```diff
@@ -14,10 +15,31 @@ def all(user, path=default_path)
...
+        def next_link_key; "nextLink"; end
+
+        def collect_items(user, base_uri)
+          next_uri = base_uri
+          item_collection = []
+
+          while next_uri
```
Comment (GreyKn) [inline lib/legato/management/finder.rb:32]: I wonder if we shouldn't make this an option, since it technically changes the output of the method

**Comment by GreyKn** [comment]
Couple minor comments, otherwise lgtm if the existing tests pass

File: lib/legato/management/finder.rb (hunk 1 above)
Comment (bfeigin) [inline lib/legato/management/finder.rb]: changed

File: lib/legato/management/finder.rb (hunk 2 above)
Comment (bfeigin) [inline lib/legato/management/finder.rb:32]: I can't really see a reason why anyone would not want all of the profiles available, I guess a maximum return size could be added but it's not the behavior we want anyway.  For the actual pr to master that seems like a reasonable behavior with something like `:infinity` being the default

**Comment by bfeigin** [comment]
@GreyKn comments addressed LG?

**Comment by GreyKn** [comment]
lgtm

**Pull request merged**